### ISSUE TITLE
Postgres-ha does not deploy using values-production.yaml

### ISSUE THREAD

**Issue by nungster** [bitnami/charts#2737]
**Which chart**:
postgres-ha

**Describe the bug**
Tried to deploy chart to Azure using the values-production.yaml.  Node 0 will come up with both metrics and postgresql containers running.  Node 1 metrics will come up but the postgresql container will timeout waiting on primary.  Primary is not accessible.

**To Reproduce**
helm install my-release bitnami/postgresql-ha -f helm-values.yaml --namespace my-namespace

**Expected behavior**
All replicas run and not crash

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.2.1", GitCommit:"fe51cd1e31e6a202cba7dead9552a6d418ded79a", GitTreeState:"clean", GoVersion:"go1.13.10"}```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-16T23:35:15Z", GoVersion:"go1.14.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.7", GitCommit:"e2d9f8479783020904aba3de7499a49be6c75ebd", GitTreeState:"clean", BuildDate:"2020-04-09T02:31:15Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}```

**Additional context**
I have found that if I disable the NetworkPolicy, all containers come up.  I am suspecting that some basic labels are needed for the production yaml to work.

**Comment by nungster** [comment]
I got my chart to deploy using the production values.  There should be an entry in the values file if a network policy is applied as in the case for the production values.
The postgresql pod needs to have a value for podLabels to label the pod as a client.
 
At line 130
podLabels:
    <deployment name>-postgresql-ha-client: "true"

**Comment by miguelaeh** [comment]
Hi @nungster ,
I am glad you solved the issue.
Thank you for sharing your solution!

**Comment by s12chung** [comment]
I feel like this label should be defaulted in the helm chart....

**Comment by miguelaeh** [comment]
Hi @s12chung ,
could you explain why do you think it should be set as a default parameter so we can evaluate it and change it if necessary?

**Comment by s12chung** [comment]
once you use these values in `production-values.yaml`:

```yaml
networkPolicy:
  enabled: true
  allowExternal: false
```

pgpool will lose connection to the nodes. the second node will never connect to pgpool. you'll get something like:

```
[2020-01-29 09:23:57] [ERROR] repmgr extension not found on this node
```

and then to figure out why, you have to look into the chart source and find what the values do, which creates a `NetworkPolicy`, then figure out the label is missing. then set the label to the proper pod by digging into the chart.

basically, without setting the `[deployment name]-postgresql-ha-client: "true"` label on the proper pods when the above values are yet, forces the chart user to basically go through this debugging process.

**Comment by miguelaeh** [comment]
Hi @s12chung ,
Thank you for the clarification. 
Then I would say it should be enough to modify the `values-production.yaml` that is where we set the `networkPolicy` to `enabled` but maintaining the `values.yaml` like it is currently. Do you agree?

**Comment by rafariossaa** [comment]
Hi @s12chung,
  It would be great if you do the PR! I'd be glad to review it

**Comment by s12chung** [comment]
have some release dates this month. i'll try to get back to this in 4 weeks.

**Comment by miguelaeh** [comment]
Hi @s12chung ,
I agree, I think we should add the label
```
{{ template "postgresql-ha.fullname" . }}-client: "true"
```
to the pods when network policy is enabled and it should solve the issue.
